python-frontend/src/test/java/org/sonar/python/types/TypeShedTest.java

```diff
@@ -67,6 +74,19 @@ public void str() {
     // python 3.9 support
     assertThat(strClass.resolveMember("removeprefix")).isNotEmpty();
     assertThat(strClass.resolveMember("removesuffix")).isNotEmpty();
+
+    setPythonVersion(PythonVersion.fromString("3.8"));
+    strClass = TypeShed.typeShedClass("str");
+    assertThat(strClass.superClasses()).extracting(Symbol::kind, Symbol::name).containsExactlyInAnyOrder(tuple(Kind.CLASS, "Sequence"));
+    assertThat(strClass.resolveMember("removeprefix")).isEmpty();
+    assertThat(strClass.resolveMember("removesuffix")).isEmpty();
+
+    setPythonVersion(PythonVersion.fromString("2.7"));
+    strClass = TypeShed.typeShedClass("str");
+    assertThat(strClass.superClasses()).extracting(Symbol::kind, Symbol::name)
+      .containsExactlyInAnyOrder(tuple(Kind.CLASS, "Sequence"), tuple(Kind.CLASS, "basestring"));
+
+    setPythonVersion(PythonVersion.allVersions());
```

Comment:
I guess that's needed to avoid polluting other tests?
Is there an alternative to this (like using `BeforeEach` or something like that)? I guess not without performance impact, but I don't like that the test somehow needs to do the cleanup itself.

Reply:
Yeah I don't like that either.
However I initially tried with `BeforeEach` and it had quite a significant performance impact and IMO it's not really worth it.
Wdyt?

Reply:
Indeed I assumed that would be the issue. I think it's fine as it is though, not ideal, but we know why we do it that way.

Reply:
I will add a comment so that we won't forget